src/main/java/picard/arrays/illumina/ExtendedIlluminaManifestRecord.java

```diff
@@ -0,0 +1,476 @@
+            log.warn("Extension base conflict: " + extensionBaseWithInsertion + ", " + extensionBaseWithoutInsertion);
+            return;
+        }
+        // Now determine whether the reference with insertion matches sequence with insertion or reference without matches sequence without.
+        // Use that to determine whether the insertion in the record exists in reference (in which case we are testing for a deletion)
+        //                 of if the insertion in the record does not exist in reference (in which case we are testing for the insertion)
+        final String referenceWithoutInsertion = getSequenceAt(refFile, record.getChr(),
+                record.getPosition() - sequenceBeforeInsertion.length(),
+                record.getPosition() + sequenceAfterInsertion.length() - 1);
+        final String referenceWithInsertion = getSequenceAt(refFile, record.getChr(),
+                record.getPosition() - sequenceBeforeInsertion.length(),
+                record.getPosition() + insertion.length() + sequenceAfterInsertion.length() - 1);
+        boolean insFound    = referenceWithInsertion.equals(sequenceWithInsertion);
+        boolean insNotFound = referenceWithoutInsertion.equals(sequenceWithoutInsertion);
+        if (insFound != insNotFound) {
+            // One or the other condition
+            // Whether the condition is an insertion or deletion is only reflected in the refAllele.
+            snpRefAllele = insFound ? baseBeforeInsertion + insertion : baseBeforeInsertion;
+            if (isAorT(extensionBaseWithInsertion)) {
+                snpAlleleA = baseBeforeInsertion + insertion;
+                snpAlleleB = baseBeforeInsertion;
+            } else {        // extensionBaseWithInsertion is C or G
+                snpAlleleA = baseBeforeInsertion;
+                snpAlleleB = baseBeforeInsertion + insertion;
+            }
+        } else {
+            flag = Flag.INDEL_SEQ_MISMATCH;
+            log.warn("Error in indel processing.  Record at " + record.getChr() + ":" + record.getPosition() + " " + record.getSnp() + " src strand:" + strand + " probe strand:" + alleleAProbeSeqStrand + " " + flag.toString());
+            log.warn("Cannot match reference and sequence with or without insertion (or matches both conditions...).");
+            log.warn("Reference with Insertion:    " + referenceWithInsertion);
+            log.warn("Sequence with Insertion:     " + sequenceWithInsertion);
+            log.warn("Sequence without Insertion:  " + sequenceWithoutInsertion);
+            log.warn("Reference without Insertion: " + referenceWithoutInsertion);
+            return;
+        }
+
+        // correcting indel coordinates for VCF format.
+        b37Pos = b37Pos - 1;
+    }
+
+    private boolean validateAlleleABBases(String allele1, String allele2) {
+        return ((isAorT(allele1) && isCorG(allele2)) || (isCorG(allele1) && isAorT(allele2)));
```

Comment:
Is htsjdk.samtools.util.SequenceUtil#readBaseMatchesRefBaseWithAmbiguity doing the same thing as this?

Reply:
I think so...

Reply:
This block was all ripped out.  It will appear in a future PR when we pull over the CreateExtendedIlluminaManifest functionality.  I have pulled the code review suggestions here and put them back into picard private.